.github/workflows/playground.yml

```diff
@@ -0,0 +1,59 @@
+name: Build playground
+on:
+  push:
+    branches: [main]
+  pull_request:
+  merge_group:
+
+# Cancel any in-flight jobs for the same PR/branch so there's only one active
+# at a time
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
```

Comment:
Mind adding this to the `main.yml` file as well? The final "gate everything here" job could gate on the `build` job here and the `deploy` job could continue to depend on `main`. Otherwise though this looks great!

I learned about `actions/{configure-pages,upload-pages-artifact,deploy-pages}` here, I didn't know what those were before!

Reply:
I'm not totally sure what you're asking here - are you suggesting getting rid of `playground.yml` and moving both of these new jobs into `main.yml`? I was aiming to have this run after commits land on `main`, which that file doesn't (appear to) do. But I guess if everything goes through the merge queue it ends up being roughly equivalent?

Reply:
> Do you know if it would be difficult to split it up like that?

Just looked and the answer appears to be that yes, it's (a little) difficult. [Artifacts](https://docs.github.com/en/actions/using-workflows/storing-workflow-data-as-artifacts) are meant to be used within a single workflow. To make this work, you'd need to download and re-upload them in the publish job - there doesn't appear to be a (supported) way to publish an artifact from a different run.

You're already [downloading artifacts from a previous run in `publish.yml`](https://github.com/bytecodealliance/wasm-tools/blob/10d2e21b2171cf2a2227f9de04ac89985042a5f0/.github/workflows/publish.yml#L61-L72), but in trying it on my own fork I couldn't immediately get [that](https://github.com/bakkot-org/wasm-tools/blob/179bf220518ff2690948ce79cd10837d15e4cc14/.github/workflows/playground.yml#L51-L84) to work - it's failing to find the `run_id`, despite that working in this project, and I don't have enough experience or interest in GitHub actions to debug it further.